CI: Move AppImage runner to a docker container.

carlosperate opened the pull request:
Let's start with an "easy" container that has already Python 3.8 set up, and when this is working we can move to an older Ubuntu container.

carlosperate commented:
@tmontes there are issues running FUSE (needed by AppImage) inside a docker container, so I've opened this issue in pup: https://github.com/mu-editor/pup/issues/201

tmontes commented:
UPDATE:

* `pup` 1.0.0a16 just released -- should work under Docker.
* Trigged a rerun of the **Build AppImage** job.

...let's see how it goes!

tmontes commented:
> UPDATE:
> 
>     * `pup` 1.0.0a16 just released -- should work under Docker.
> 
>     * Trigged a rerun of the **Build AppImage** job.
> 
> 
> ...let's see how it goes!

SUCCESS! :-)

carlosperate commented:
Awesome, thanks @tmontes!

I gave it a quick test in Ubuntu 16.04, and the app launches which is great! But some of the wheels downloaded are for newer Linux distribution releases, it fails trying to install them.

I'll start updating the docker image to use Ubuntu 16.04 and see if that works in old and new distributions.

carlosperate commented on Makefile:
```diff
@@ -121,7 +121,7 @@ macos: check
 	ls -la ./build/pup/
 	ls -la ./dist/
 
-linux: check
```
Temporarily removing running the tests, as it currently fails due to this unrelated issue: https://github.com/mu-editor/mu/issues/2091

carlosperate replied:
After PR https://github.com/mu-editor/mu/pull/2238 was merged and this PR rebased I was able to remove this workaround.

carlosperate commented:
I've tested the AppImage from https://github.com/mu-editor/mu/actions/runs/2315395087 in Ubuntu 16.04 and 20.04 and seems to be working 🎉

tmontes commented:
> I've tested the AppImage from https://github.com/mu-editor/mu/actions/runs/2315395087 in Ubuntu 16.04 and 20.04 and seems to be working 🎉

In a rush, but runs on CentOS 7, too: REPL + turtle graphics quick test works.

carlosperate commented:
Also tested in Ubuntu 22.04 and it has the same issues as the previous AppImage, so no regressions there: https://github.com/mu-editor/mu/issues/2223

carlosperate commented:
The workflow was installing Python 3.8 into Ubuntu 16.04 by downloading one of the Python standalong builds and "installing it", and then installing other required packages. So decided to make a custom Docker image based on 16.04 with with everything pre-installed. This reduced the time it takes to build, and it also reduces a bit how many times we are fetching the Python standalone builds (although pup still needs to do that).

The new Docker image is also hosted in the GitHub package registry: https://github.com/mu-editor/docker-mu-appimage

carlosperate commented:
Now that https://github.com/mu-editor/mu/issues/2091 has been fixed via PR https://github.com/mu-editor/mu/pull/2238, I've rebased and removed the workaround applied to this branch, so I can move this PR from draft to ready.

carlosperate commented:
@tmontes any objections to merge this?
